src/Utilities.cpp

```diff
@@ -547,54 +547,43 @@ double getSemiMinorRadius(json isd, csm::WarningList *list) {
   return radius;
 }
 
-std::vector<double> getTransverseDistortionX(json isd, csm::WarningList *list) {
+std::vector<double> getDistortionCoeffs(json isd, csm::WarningList *list) {
   std::vector<double> coefficients;
   try {
-    coefficients = isd.at("optical_distortion").at("transverse").at("x").get<std::vector<double>>();
-    coefficients.resize(10, 0.0);
-  }
-  catch (...) {
-    if (list) {
-      list->push_back(
-        csm::Warning(
-          csm::Warning::DATA_NOT_AVAILABLE,
-          "Could not parse the transverse distortion model X coefficients.",
-          "Utilities::getTransverseDistortionX()"));
-    }
-  }
-  return coefficients;
-}
+    std::vector<double> coefficientsX, coefficientsY;
 
-std::vector<double> getTransverseDistortionY(json isd, csm::WarningList *list) {
-  std::vector<double> coefficients;
-  try {
-    coefficients = isd.at("optical_distortion").at("transverse").at("y").get<std::vector<double>>();
-    coefficients.resize(10, 0.0);
+    coefficientsX = isd.at("optical_distortion").at("transverse").at("x").get<std::vector<double>>();
+    coefficientsX.resize(10, 0.0);
+
+    coefficientsY = isd.at("optical_distortion").at("transverse").at("y").get<std::vector<double>>();
+    coefficientsY.resize(10, 0.0);
+
+    coefficients = coefficientsX;
+
+    coefficients.insert(coefficients.end(), coefficientsY.begin(), coefficientsY.end());
+    return coefficients;
   }
   catch (...) {
     if (list) {
       list->push_back(
         csm::Warning(
           csm::Warning::DATA_NOT_AVAILABLE,
-          "Could not parse the transverse distortion model Y coefficients.",
-          "Utilities::getTransverseDistortionY()"));
+          "Could not parse a transverse distortion model coefficients.",
+          "Utilities::getDistortion()"));
     }
   }
-  return coefficients;
-}
-
-std::vector<double> getRadialDistortion(json isd, csm::WarningList *list) {
-  std::vector<double> coefficients;
   try {
     coefficients = isd.at("optical_distortion").at("radial").at("coefficients").get<std::vector<double>>();
+
+    return coefficients;
   }
   catch (...) {
     if (list) {
       list->push_back(
         csm::Warning(
           csm::Warning::DATA_NOT_AVAILABLE,
           "Could not parse the radial distortion model coefficients.",
-          "Utilities::getRadialDistortion()"));
+          "Utilities::getDistortion()"));
```

Comment:
This scheme will return a warning if there are radial but not transverse distortion coefficients. That's a problem, because the current logic says an ISD cannot be used if any warnings are returned. Is it possible to return a warning only if neither could be found?